src/analysis/convex/basic.lean

```diff
@@ -625,85 +671,217 @@ begin
+    { intros i hi,
+      rw [finset.mem_union, finset.mem_image, finset.mem_image] at hi,
+      rcases hi with ⟨j, hj, rfl⟩|⟨j, hj, rfl⟩;
+        simp only [sum.elim_inl, sum.elim_inr]; apply_rules [hzx, hzy] } },
+  { rintros _ ⟨α, t, w, z, hw₀, hw₁, hz, rfl⟩,
+    exact t.center_mass_mem_convex_hull hw₀ (hw₁.symm ▸ zero_lt_one) hz }
+end
+
+lemma set.finite.convex_hull_eq {s : set E} (hs : finite s) :
+  convex_hull s = {x : E | ∃ (w : E → ℝ) (hw₀ : ∀ y ∈ s, 0 ≤ w y) (hw₁ : hs.to_finset.sum w = 1),
+    hs.to_finset.center_mass w id = x} :=
+begin
+  refine subset.antisymm (convex_hull_min _ _) _,
+  { intros x hx,
+    replace hx : x ∈ hs.to_finset, from finite.mem_to_finset.2 hx,
+    refine ⟨_, _, _, finset.center_mass_ite_eq _ _ _ hx⟩,
+    { intros, split_ifs, exacts [zero_le_one, le_refl 0] },
+    { rw [finset.sum_ite_eq, if_pos hx] } },
+  { rintros x y ⟨wx, hwx₀, hwx₁, rfl⟩ ⟨wy, hwy₀, hwy₁, rfl⟩
+      a b ha hb hab,
+    rw [finset.center_mass_segment _ _ _ _ hwx₁ hwy₁ _ _ hab],
+    refine ⟨_, _, _, rfl⟩,
+    { rintros i hi,
+      apply_rules [add_nonneg, mul_nonneg, hwx₀, hwy₀], },
+    { simp only [finset.sum_add_distrib, finset.mul_sum.symm, mul_one, *] } },
+  { rintros _ ⟨w, hw₀, hw₁, rfl⟩,
+    exact hs.to_finset.center_mass_mem_convex_hull (λ x hx, hw₀ _ $ finite.mem_to_finset.1 hx)
+      (hw₁.symm ▸ zero_lt_one) (λ x hx, finite.mem_to_finset.1 hx) }
+end
+
+lemma is_linear_map.convex_hull_image {f : E → F} (hf : is_linear_map ℝ f) (s : set E) :
+  convex_hull (f '' s) = f '' convex_hull s :=
+set.subset.antisymm (convex_hull_min (image_subset _ (subset_convex_hull s)) $
+  (convex_convex_hull s).is_linear_image hf)
+  (image_subset_iff.2 $ convex_hull_min
+    (image_subset_iff.1 $ subset_convex_hull _)
+    ((convex_convex_hull _).is_linear_preimage hf))
+
+lemma linear_map.convex_hull_image (f : E →ₗ[ℝ] F) (s : set E) :
+  convex_hull (f '' s) = f '' convex_hull s :=
+f.is_linear.convex_hull_image s
+
+end convex_hull
+
+/-! ### Simplex -/
+
+section simplex
+
+variables (α) [fintype α] {f : α → ℝ}
+
+/-- Standard simplex in the space of functions `α → ℝ`. -/
+def std_simplex (α : Type*) [fintype α] : set (α → ℝ) :=
+{ f | (∀ x, 0 ≤ f x) ∧ finset.univ.sum f = 1 }
+
+lemma std_simplex_eq_inter :
+  std_simplex α = (⋂ x, {f | 0 ≤ f x}) ∩ {f | finset.univ.sum f = 1} :=
+by { ext f, simp only [std_simplex, set.mem_inter_eq, set.mem_Inter, set.mem_set_of_eq] }
+
+lemma convex_std_simplex : convex (std_simplex α) :=
+begin
+  refine λ f g hf hg a b ha hb hab, ⟨λ x, _, _⟩,
+  { apply_rules [add_nonneg, mul_nonneg, hf.1, hg.1] },
+  { erw [finset.sum_add_distrib, ← finset.smul_sum, ← finset.smul_sum, hf.2, hg.2,
+      smul_eq_mul, smul_eq_mul, mul_one, mul_one],
+    exact hab }
+end
+
+variable {α}
+
+lemma ite_eq_mem_std_simplex (i : α) : (λ j, ite (i = j) (1:ℝ) 0) ∈ std_simplex α :=
+⟨λ j, by simp only []; split_ifs; norm_num, by rw [finset.sum_ite_eq, if_pos (finset.mem_univ _)] ⟩
+
+lemma convex_hull_std_basis_eq :
```

Comment:
There is a standard basis somewhere in the linear algebra library. Does it relate to the notion you are using?

Reply:
When I was writing this, I failed to reuse `std_basis` here. I'll try to look into this once more.

Reply:
Renamed. `std_basis i` is actually the linear embedding of `φ i` into `Πi, φ i`. I'm not sure that `std_basis` is the best name in this case.